db/db_impl.cc

```diff
@@ -2967,12 +2978,10 @@ Status DBImpl::GetLatestSequenceForKey(SuperVersion* sv, const Slice& key,
       ROCKS_LOG_ERROR(immutable_db_options_.info_log,
                       "Unexpected status returned from Version::Get: %s\n",
                       s.ToString().c_str());
-
-      return s;
     }
   }
 
-  return Status::OK();
+  return s;
```

Comment:
The return status here may be NotFound

Reply:
There's a call site in transaction_util.cc but interestingly it don't quite rely on the return code so it was fine and didn't reveal the issue.. Thanks for the good catch!